api/v1beta1/designatecentral_types.go

```diff
@@ -54,7 +54,7 @@ type DesignateCentralSpec struct {
 
 	// +kubebuilder:validation:Required
 	// Secret containing OpenStack password information for designate DesignateDatabasePassword, AdminPassword
-	Secret string `json:"secret,omitempty"`
```

Comment:
so, ginkgolinter does not like the omitempty tag?

Reply:
I think our `operator-lint` linter should be flagging this.  Perhaps we don't have that linter here though.  Anyhow, `Required` and `omitempty` are mutually exclusive, as `omitempty` is basically synonymous with `Optional`.

Reply:
Yeah make-operator-lint failed with below error:

make-operator-lint.......................................................Failed
- hook id: make-operator-lint
- exit code: 2

test -f go.work || go work init
go work use .
go work use ./api
go work sync
GOBIN=/home/bhagyashris/src/next_gen/operators/designate-operator/bin go install github.com/gibizer/operator-lint@v0.1.0
go vet -vettool=/home/bhagyashris/src/next_gen/operators/designate-operator/bin/operator-lint ./... ./api/...
# github.com/openstack-k8s-operators/cinder-operator/api/v1beta1
api/v1beta1/designateagent_types.go:57:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
api/v1beta1/designatecentral_types.go:57:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
api/v1beta1/designatemdns_types.go:60:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
api/v1beta1/designateproducer_types.go:60:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
api/v1beta1/designatesink_types.go:60:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
api/v1beta1/designateworker_types.go:60:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
make: *** [Makefile:300: operator-lint] Error 2

and to resolve that i have removed omitempty.